src/textual/dom.py

```diff
@@ -596,3 +600,8 @@ def has_pseudo_class(self, *class_names: str) -> bool:
 
     def refresh(self, *, repaint: bool = True, layout: bool = False) -> None:
         pass
+
+
+def _focus_sort_key(widget: Widget) -> tuple[int, int]:
```

Comment:
I have a feeling we can optimize this with a combination of `attrgetter` and a new property.

It's not crazy to imagine a layout with 1000s of widgets.

I'm thinking that widget could grow a `_focus_sort_key` attribute. which return a tuple of (y, x), and we can get that with `attrgetter`

Reply:
I've made this change, and also moved `focusable_children` down to `Widget`. It's only used in the context of `Widget` types elsewhere, and in calculating it requires knowing the `virtual_region`, which doesn't exist at the `DOMNode` level.